pkg/controllers/ipam/ipam.go

```diff
@@ -176,24 +329,50 @@ func (i *IPAM) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 	clone := eip.DeepCopy()
 
 	if err = i.updateEip(clone); err != nil {
-		i.Client.Status().Update(context.Background(), clone)
+		i.Client.Status().Update(ctx, clone)
```

Comment:
I think that once an error occurs here, there should be no need to update the status of EIP.